### ISSUE TITLE
[3.4] app.import type: 'test' + transformation is broken

### ISSUE THREAD

**Issue by lennyburdette** [ember-cli/ember-cli#8039]
Reproduction: https://github.com/lennyburdette/ember-3-4-import-bug

This import works fine, allowing `import sinon from 'sinon'`.

```js
app.import('node_modules/sinon/pkg/sinon.js', {
  using: [{ transformation: 'amd', as: 'sinon' }]
});
```

But if I add `type: "test"` so that the code is appended to test-support.js, it appears in the output without the transform.

It looks like `applyCustomTransforms()` is called only in [`processAppAndDependencies()`](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/default-packager.js#L252), not in [`packageTestFiles()`](https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/default-packager.js#L1098).

cc @twokul @rwjblue

**Comment by rwjblue** [comment]
Thanks for reporting @lennyburdette! I totally agree with your assessment.

**Comment by rwjblue** [comment]
Fixed by #8047

**Comment by lennyburdette** [comment]
Thanks!!

**Comment by rwjblue** [comment]
fix released in v3.4.3
